src/test/java/org/folio/rest/impl/EndToEndTest.java

```diff
@@ -0,0 +1,353 @@
+  private static final String FILES_FOR_UPLOAD_DIRECTORY = "endToEndTestFiles/";
+  private static final String UPLOAD_URL = "/upload";
+  private static final String FILE_SYSTEM_DESTINATION = "fileSystem";
+  private static final String SRS_RESPONSE_FILE_NAME = "srsResponse.json";
+  private static final String FILE_WITH_NON_EXITING_UUID = "InventoryUUIDsNonExiting.csv";
+  private static final String FILE_WITH_TWO_BATCHES_OF_UUIDS = "InventoryUUIDsTwoBatches.csv";
+  private static final String EMPTY_FILE = "InventoryUUIDsEmptyFile.csv";
+  private static final String FILE_WITH_ONE_BATCH_OF_UUIDS = "InventoryUUIDsOneBatch.csv";
+  private static final String DASH = "-";
+  private static final String DATE_TIME_REGEX = "[0-9]{14}";
+  private static final String MRC_EXTENSION = "mrc";
+  private static final long TIMER_DELAY = 1000L;
+
+  private static SourceRecordStorageClient mockSourceRecordStorageClient = Mockito.mock(SourceRecordStorageClient.class);
+  private static ExportStorageService mockExportStorageService = Mockito.mock(ExportStorageService.class);
+
+  @Autowired
+  private JobExecutionDao jobExecutionDao;
+  @Autowired
+  private FileDefinitionDao fileDefinitionDao;
+
+
+  public EndToEndTest() {
+    Context vertxContext = vertx.getOrCreateContext();
+    SpringContextUtil.init(vertxContext.owner(), vertxContext, EndToEndTest.TestConfig.class);
+    SpringContextUtil.autowireDependencies(this, vertxContext);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    FileUtils.deleteDirectory(new File(STORAGE_DIRECTORY_PATH));
+  }
+
+  @Test
+  public void shouldReturn_204Status_forHappyPathExport(TestContext context) throws IOException, InterruptedException {
+    Async async = context.async();
+
+    //given
+    FileDefinition uploadedFileDefinition = givenUploadFile(FILE_WITH_ONE_BATCH_OF_UUIDS);
+
+    //when
+    ExportRequest exportRequest = getExportRequest(uploadedFileDefinition);
+    Response response = RestAssured.given()
+      .spec(jsonRequestSpecification)
+      .body(JsonObject.mapFrom(exportRequest).encode())
+      .when()
+      .post(EXPORT_URL);
+
+    // then
+    context.assertEquals(HttpStatus.SC_NO_CONTENT, response.getStatusCode());
+    async.complete();
+  }
+
+  @Test
+  public void shouldExportFileWithRecords_whenExportInOneBatch(TestContext context) throws IOException, InterruptedException {
+    Async async = context.async();
+
+    //given
+    givenSetSourceStorageMockToReturnRecords();
+    ArgumentCaptor<FileDefinition> fileExportDefinitionCaptor = givenCaptureFileExportDefinition();
+    FileDefinition uploadedFileDefinition = givenUploadFile(FILE_WITH_ONE_BATCH_OF_UUIDS);
+
+    //when
+    ExportRequest exportRequest = getExportRequest(uploadedFileDefinition);
+    RestAssured.given()
+      .spec(jsonRequestSpecification)
+      .body(JsonObject.mapFrom(exportRequest).encode())
+      .when()
+      .post(EXPORT_URL);
+
+    // then
+    vertx.setTimer(1000L, handler -> {
```

Comment:
I don't think using a timer is a good idea. There is a chance that the tests can fail in other environments

Reply:
I don't know other options, without timer test will fail definitely, the same approach is used for mod-inventory. It's possible to use Awaitlity library, but it can take some time to investigate how to use it in our cases

Reply:
Maybe at least we can set delay longer